From: Debian/Kubuntu Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
Date: Fri, 11 Nov 2016 09:45:34 +0100
Subject: _sddm_systemctl

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,8 +143,8 @@ if(SYSTEMD_FOUND)
     endif()
 
     set(MINIMUM_VT 1)
-    set(HALT_COMMAND "/usr/bin/systemctl poweroff")
-    set(REBOOT_COMMAND "/usr/bin/systemctl reboot")
+    set(HALT_COMMAND "/bin/systemctl poweroff")
+    set(REBOOT_COMMAND "/bin/systemctl reboot")
 else()
     set(SYSTEMD_FOUND 0)
 endif()
