--- a/src/common/Configuration.h
+++ b/src/common/Configuration.h
@@ -45,7 +45,7 @@
         //      Name   Entries (but it's a regular class again)
         Section(Theme,
             Entry(ThemeDir,            QString,     _S(DATA_INSTALL_DIR "/themes"),             _S("Theme directory path"));
-            Entry(Current,             QString,     _S("maui"),                                 _S("Current theme name"));
+            Entry(Current,             QString,     _S("breeze"),                                 _S("Current theme name"));
             Entry(FacesDir,            QString,     _S(DATA_INSTALL_DIR "/faces"),              _S("Face icon directory\n"
                                                                                                    "The files should be in username.face.icon format"));
             Entry(CursorTheme,         QString,     QString(),                                  _S("Cursor theme"));
