From: Maximiliano Curia <maxy@gnuservers.com.ar>
Date: Fri, 11 Nov 2016 10:23:38 +0100
Subject: handle reading from a closed file handle

---
 src/daemon/XorgDisplayServer.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/src/daemon/XorgDisplayServer.cpp
+++ b/src/daemon/XorgDisplayServer.cpp
@@ -193,6 +193,12 @@ namespace SDDM {
                 return false;
             }
             QByteArray displayNumber = readPipe.readLine();
+            if (displayNumber.isEmpty()) {
+                // the file descriptor was closed without a display number,
+                // the X has probably died.
+                close(pipeFds[0]);
+                return false;
+            }
             displayNumber.prepend(QByteArray(":"));
             displayNumber.remove(displayNumber.size() -1, 1); //trim trailing whitespace
             m_display = QString::fromLocal8Bit(displayNumber);
