diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf3fb70..5d4bc62 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,14 +98,14 @@ if(SYSTEMD_FOUND)
     add_definitions(-DHAVE_SYSTEMD)
     set(CMAKE_AUTOMOC_MOC_OPTIONS -DHAVE_SYSTEMD)
 
-    pkg_check_modules(JOURNALD "libsystemd-journal")
+    pkg_check_modules(JOURNALD "libsystemd")
 
     if(ENABLE_JOURNALD)
         if(JOURNALD_FOUND)
             add_definitions(-DHAVE_JOURNALD)
             set(CMAKE_AUTOMOC_MOC_OPTIONS -DHAVE_JOURNALD)
         else()
-            message(WARNING "Disable journald support for lack of libsystemd-journal")
+            message(WARNING "Disable journald support for lack of libsystemd")
         endif()
     endif()
 
