From: Maximiliano Curia <maxy@gnuservers.com.ar>
Date: Sun, 12 Mar 2017 17:26:05 +0100
Subject: Regression fix, disable HiDPI by default

---
 src/common/Configuration.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/common/Configuration.h
+++ b/src/common/Configuration.h
@@ -42,7 +42,7 @@ namespace SDDM {
         Entry(Numlock,             NumState,    NUM_NONE,                                       _S("Initial NumLock state. Can be on, off or none.\n"
                                                                                                    "If property is set to none, numlock won't be changed\n"
                                                                                                    "NOTE: Currently ignored if autologin is enabled."));
-        Entry(EnableHiDPI,         bool,        true,                                           _S("Enable Qt's automatic high-DPI scaling"));
+        Entry(EnableHiDPI,         bool,        false,                                          _S("Enable Qt's automatic high-DPI scaling"));
         Entry(InputMethod,         QString,     QString(),                                      _S("Input method module"));
         //  Name   Entries (but it's a regular class again)
         Section(Theme,
