diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf3fb70..e0306dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,8 +112,8 @@ if(SYSTEMD_FOUND)
     execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=systemdsystemunitdir systemd OUTPUT_VARIABLE SYSTEMD_SYSTEM_UNIT_DIR)
     string(REGEX REPLACE "[ \t\n]+" "" SYSTEMD_SYSTEM_UNIT_DIR ${SYSTEMD_SYSTEM_UNIT_DIR})
     set(MINIMUM_VT 7)
-    set(HALT_COMMAND "/usr/bin/systemctl poweroff")
-    set(REBOOT_COMMAND "/usr/bin/systemctl reboot")
+    set(HALT_COMMAND "/bin/systemctl poweroff")
+    set(REBOOT_COMMAND "/bin/systemctl reboot")
 else()
     set(SYSTEMD_FOUND 0)
     set(MINIMUM_VT 7)
