Description: Replace systemd-journal with systemd.
Author: Dimitri John Ledkov <xnox@ubuntu.com>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,7 +98,7 @@
     add_definitions(-DHAVE_SYSTEMD)
     set(CMAKE_AUTOMOC_MOC_OPTIONS -DHAVE_SYSTEMD)
 
-    pkg_check_modules(JOURNALD "libsystemd-journal")
+    pkg_check_modules(JOURNALD "libsystemd")
 
     if(ENABLE_JOURNALD)
         if(JOURNALD_FOUND)
